core/src/syscall/unconstrained.rs

```diff
@@ -23,7 +23,7 @@ impl Syscall for SyscallEnterUnconstrained {
             record: std::mem::take(&mut ctx.rt.record),
             op_record: std::mem::take(&mut ctx.rt.memory_accesses),
         };
-        1
+        None
```

Comment:
I think this needs to be 1, that's how the program knows to enter unconstrained mode. But in the VM it actually needs to be constrained to return 0. Also, the exit syscall needs to return 0.